### PR TITLE
Trap Tweaks & Fixes

### PR DESCRIPTION
**Pull request by Geevies**
* Animal traps now work based off mob size, instead of having a specific mapped in list for animals it can or cannot capture.
* Added new examine text to traps that help explain how to use them.
* Added a verb to deploy / undeploy traps.
* Tweaked a couple interactions with traps, mechanically and grammatically.
* Fixed a lot of bugs and issues with traps.

- [x] Finish looking over the file and removing the bad.
- [x] Clear out the TO-DO's I left in the file.
- [x] Add ability to examine the item to figure out how to use the item.
- [x] Test changes.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by Geevies** [comment]
This is ready for review, but I'd like a test merge just to let players test it out and see if they can catch any weird edge case bugs I missed during testing.

**Review by Arrow768 — APPROVED** [review]

File: code/game/objects/items/weapons/traps.dm [hunk 1]
```diff
@@ -32,38 +56,26 @@
...
-			user.visible_message(
-				SPAN_NOTICE("\The [user] frees \the [buckled] from \the [src]."),
-				SPAN_NOTICE("You free \the [buckled] from \the [src].")
-				)
-			unbuckle()
-			anchored = FALSE
+	if(!buckled || !can_use(user))
+		return
+
+	user.visible_message("<b>[user]</b> begins freeing \the [buckled] from \the [src]...", SPAN_NOTICE("You carefully begin to free \the [buckled] from \the [src]..."), SPAN_NOTICE("You hear metal creaking."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:73]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 2]
```diff
@@ -32,38 +56,26 @@
...
+
+	if(do_after(user, time_to_escape))
+		user.visible_message("<b>[user]</b> frees \the [buckled] from \the [src].", SPAN_NOTICE("You free \the [buckled] from \the [src]."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:76]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 3]
```diff
@@ -32,38 +56,26 @@
 			anchored = TRUE
 
 /obj/item/trap/proc/deploy(mob/user)
-	user.visible_message(
-		SPAN_WARNING("\The [user] starts to deploy \the [src]."),
-		SPAN_WARNING("You begin deploying \the [src]!"),
-		SPAN_WARNING("You hear the slow creaking of a spring.")
-		)
+	user.visible_message("<b>[user]</b> starts deploying \the [src]...", SPAN_NOTICE("You begin deploying \the [src]!"), SPAN_WARNING("You hear the slow creaking of a spring."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:59]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 4]
```diff
@@ -32,38 +56,26 @@
...
 
 	if(do_after(user, 5 SECONDS))
-		user.visible_message(
-			SPAN_WARNING("\The [user] deploys \the [src]."),
-			SPAN_WARNING("You deploy \the [src]!"),
-			SPAN_WARNING("You hear a latch click loudly.")
-			)
-
+		user.visible_message("<b>[user]</b> deploys \the [src].", SPAN_WARNING("You deploy \the [src]!"), SPAN_WARNING("You hear a latch click loudly."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:62]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 5]
```diff
@@ -76,19 +88,16 @@
 	..()
 
 /obj/item/trap/proc/disarm_trap(var/mob/user)
-	user.visible_message(SPAN_NOTICE("\The [user] starts to disarm \the [src]..."), SPAN_NOTICE("You begin disarming \the [src]..."), SPAN_WARNING("You hear a latch click followed by the slow creaking of a spring."))
+	user.visible_message("<b>[user]</b> starts disarming \the [src]...", SPAN_NOTICE("You begin disarming \the [src]..."), SPAN_WARNING("You hear a latch click followed by the slow creaking of a spring."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:91]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 6]
```diff
@@ -76,19 +88,16 @@
...
 	if(do_after(user, 6 SECONDS))
-		user.visible_message(SPAN_NOTICE("\The [user] disarms \the [src]!"), SPAN_NOTICE("You disarm \the [src]!"))
+		user.visible_message("<b>[user]</b> disarms \the [src]!", SPAN_NOTICE("You disarm \the [src]!"))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:93]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 7]
```diff
@@ -348,91 +575,85 @@
...
 
-		if (G.state == GRAB_PASSIVE || G.state == GRAB_UPGRADING)
-			to_chat(user, SPAN_NOTICE("You need a better grip on \the [M]!"))
+		if (grab.state == GRAB_PASSIVE || grab.state == GRAB_UPGRADING)
+			to_chat(user, SPAN_NOTICE("You need a better grip on \the [capturing_mob]!"))
 			return
 
-		user.visible_message(SPAN_NOTICE("[user] starts putting [M] into \the [src]."),
-								SPAN_NOTICE("You start putting [M] into \the [src]."))
+		user.visible_message("<b>[user]</b> starts putting \the [capturing_mob] into \the [src].", SPAN_NOTICE("You start putting \the [capturing_mob] into \the [src]."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:614]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 8]
```diff
@@ -348,91 +575,85 @@
...
 		if(!T)
-			to_chat(user, SPAN_WARNING("There is nothing to secure [src] to!"))
+			to_chat(user, SPAN_WARNING("There is nothing to secure \the [src] to!"))
 			return
 
-		user.visible_message(SPAN_NOTICE("[user] is trying to [anchored ? "un" : "" ]secure \the [src]!"),
-								SPAN_NOTICE("You are trying to [anchored ? "un" : "" ]secure \the [src]!"))
-		var/sound_to_play = pick(list('sound/items/Screwdriver.ogg', 'sound/items/Screwdriver2.ogg'))
-		playsound(src.loc, sound_to_play, 50, 1)
+		user.visible_message("<b>[user]</b> starts [anchored ? "un" : "" ]securing \the [src]...", SPAN_NOTICE("You start [anchored ? "un" : "" ]securing \the [src]..."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:647]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 9]
```diff
@@ -348,91 +575,85 @@
...
 
-		if(attacking_item.use_tool(src, user, 30, volume = 50))
+		if(attacking_item.use_tool(src, user, 3 SECONDS, volume = 50))
 			density = !density
 			anchored = !anchored
-			user.visible_message(SPAN_NOTICE("[user] [anchored ? "" : "un" ]secures \the [src]!"),
-									SPAN_NOTICE("You [anchored ? "" : "un" ]secure \the [src]!"))
+			if(!anchored)
+				deployed = FALSE
+			user.visible_message("<b>[user]</b> [anchored ? "" : "un" ]secures \the [src]!", SPAN_NOTICE("You [anchored ? "" : "un" ]secure \the [src]!"))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:654]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 10]
```diff
@@ -453,52 +674,37 @@
...
-								SPAN_NOTICE("You pass through \the [src] without triggering it.")
-				)
-			else
-				user.forceMove(loc)
-				user.visible_message(SPAN_WARNING("[user] accidentally triggers \the [src]!"),
-								SPAN_WARNING("You accidentally trigger \the [src]!")
-				)
-				capture(user)
+/obj/item/trap/animal/proc/pass_without_trace(var/mob/user, var/chance_to_trigger = 0)
+	user.visible_message("<b>[user]</b> tries moving around \the [src] without triggering it.", SPAN_NOTICE("You try to carefully move around \the [src] without triggering it."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:685]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 11]
```diff
@@ -453,52 +674,37 @@
...
+	if(do_after(user, 2 SECONDS, src))
+		if(chance_to_trigger && prob(chance_to_trigger))
+			user.forceMove(loc)
+			user.visible_message(SPAN_WARNING("[user] accidentally triggers \the [src]!"), SPAN_WARNING("You accidentally trigger \the [src]!"))
+			capture(user)
+		else
+			user.forceMove(loc)
+			user.visible_message("<b>[user]</b> successfully moves around \the [src] without triggering it.", SPAN_NOTICE("You successfully move around \the [src] without triggering it."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:693]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 12]
```diff
@@ -657,196 +849,29 @@
...
+	. = ..()
+	. += SPAN_NOTICE("\The [src] can be turned into a large trap by attaching twelve metal rods to it.")
+
 /obj/item/large_trap_foundation/attackby(obj/item/attacking_item, mob/user)
 	if(istype(attacking_item, /obj/item/stack/rods))
 		var/obj/item/stack/rods/O = attacking_item
 		if(O.get_amount() >= 12)
-
-			to_chat(user, SPAN_NOTICE("You are trying to add metal bars to \the [src]."))
+			user.visible_message("<b>[user]</b> starts adding metal bars to \the [src]...", SPAN_NOTICE("You start adding metal bars to \the [src]..."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:860]: SPAN_BOLD

File: code/game/objects/items/weapons/traps.dm [hunk 13]
```diff
@@ -657,196 +849,29 @@
...
 
 			if (!do_after(user, 2 SECONDS, src))
 				return
-
-			to_chat(user, SPAN_NOTICE("You add metal bars to \the [src]."))
 			O.use(12)
-			new /obj/item/trap/animal/large(src.loc)
+
+			var/obj/item/trap/animal/large/new_trap = new /obj/item/trap/animal/large(src.loc)
+			user.visible_message("<b>[user]</b> finishes constructing \the [new_trap].", SPAN_NOTICE("You finish constructing \the [new_trap]."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/traps.dm:867]: SPAN_BOLD

**Review by FluffyGhoster — CHANGES_REQUESTED** [review]

**Comment by Geevies** [comment]
!review

**Review by NonQueueingMatt — APPROVED** [review]
